bubble_sort.rb

```diff
@@ -0,0 +1,39 @@
+def bubble_sort(arr)
+  if arr.length == 1
+    return arr
+  end
+  swapped = true
+  while swapped
+    swapped = false
+    (0..arr.length-2).each do |idx|
+      if arr[idx] > arr[idx+1]
```

Comment:
This is relying on the "native" comparison for whatever you pass in. It wouldn't probably work (or at least not how you expect) if you passed in an array of some objects.

Reply:
ooh good to know